SCPSLAudioApi/AudioCore/AudioPlayerBase.cs

```diff
@@ -368,26 +404,37 @@ public virtual void Update()
 
             if (LogDebug)
                 Log.Debug($"2 {toCopy} {allowedSamples} {samplesPerSecond} {StreamBuffer.Count} {PlaybackBuffer.Length} {PlaybackBuffer.WriteHead}");
-            
+
             allowedSamples -= toCopy;
 
             while (PlaybackBuffer.Length >= 480)
             {
                 PlaybackBuffer.ReadTo(SendBuffer, (long)480, 0L);
                 int dataLen = Encoder.Encode(SendBuffer, EncodedBuffer, 480);
-                
+
                 foreach (var plr in ReferenceHub.AllHubs)
                 {
                     if (plr.connectionToClient == null || !PlayerIsConnected(plr) || (BroadcastTo.Count >= 1 && !BroadcastTo.Contains(plr.PlayerId))) continue;
-                    
+
                     plr.connectionToClient.Send(new VoiceMessage(Owner, BroadcastChannel, EncodedBuffer, dataLen, false));
                 }
             }
         }
 
+        /// <summary>
+        /// Checks whether a player connected to the server is considered fully connected or if it is a DummyPlayer.
+        /// </summary>
+        /// <param name="hub">The ReferenceHub of the player to check.</param>
+        /// <returns>True if the player is fully connected and not a DummyPlayer; otherwise, false.</returns>
         private bool PlayerIsConnected(ReferenceHub hub)
         {
-            return hub.authManager.InstanceMode == ClientInstanceMode.ReadyClient && hub.nicknameSync.NickSet;
+            return hub.authManager.InstanceMode == ClientInstanceMode.ReadyClient &&
+                   hub.nicknameSync.NickSet && 
+                   !hub.isLocalPlayer &&
+                   !string.IsNullOrEmpty(hub.authManager.UserId) &&
+                   hub.authManager.UserId != null &&
```

Comment:
The userid is checked for being null checked twice, once by IsNullOrEmpty and the second time with a null check.

Reply:
ups